TechnicalOverview.md tweaks (#512)
zkSNACKs/WasabiDoc: docs/building-wasabi/TechnicalOverview.md

Let's look at what is going on under the hood for Wasabi: what design decisions and tradeoffs the developers have made, so we can later understand where it can be improved.
After setting up Wasabi and generating a wallet, Wasabi welcomes the user with a "Load Wallet" screen.

Unlike other wallets, Wasabi has a convenient way to use multiple wallets.
Privacy-centric users may be already used to achieving coin separation this way.
However, Wasabi provides a convenient in-wallet coin separation interface too.
More information will be provided about that later on.

Since coin separation can be easily achieved without multiple wallet files, initially the developers did not plan for such a wallet management system.
Our UX design choices have naturally led us down this road.

![](/WalletManagerLoadWallet.png)

Wasabi has a status bar that shows meta-information about the state of the wallet.
To better understand the architecture of the wallet it, is helpful to go through the information provided.

The "Tor" label shows the status of the Tor daemon.
Tor is an anonymity network which Wasabi includes and runs by default in the background.

> The user can also opt to use their own Tor instance. All Internet traffic goes through Tor, and by default all this traffic stays inside the onion network.

Reviewer: ```suggestion
The user can also opt to use their own Tor instance.
All Internet traffic goes through Tor, and by default all this traffic stays inside the onion network.
```

The first iteration of Wasabi was [HiddenWallet](https://github.com/zkSNACKs/WalletWasabi/tree/hiddenwallet-v0.6).
This was a full-block SPV wallet that aimed to leverage usability without compromising privacy, through the omission of initial blockchain downloading compared to a full node.

In theory, it was a light wallet.
In practice, it was hard to compete with Bitcoin Core's micro-optimizations, and it was still painful to wait for wallet synchronization every time the wallet was opened.

Now, Wasabi Wallet users have the option of using the built-in Bitcoin Core full node, connecting to a different full node, or running the wallet in "light" mode.

[Read more about network-level Bitcoin wallet privacy here.](/using-wasabi/NetworkLevelPrivacy.md)

![](/StatusBarBackendConnected.png)

Back to Wasabi.
After loading the wallet, the user can generate a receive address.
Some important design choices were made here.
First, Wasabi had to be a Segregated Witness only wallet, so the registration of unconfirmed CoinJoin outputs into a new CoinJoin round is done to prevent malleability attacks.

> However, the developers of Wasabi decided to make the wallet native Segwit (bech32) only, not supporting wrapped Segwit.

Reviewer 2: ```suggestion
However, the developers of Wasabi decided to make the wallet native Segwit (bech32) only, not supporting wrapped SegWit.
```

In theory, it was a light wallet.
In practice, it was hard to compete with Bitcoin Core's micro-optimizations, and it was still painful to wait for wallet synchronization every time the wallet was opened.

Now, Wasabi Wallet users have the option of using the built-in Bitcoin Core full node, connecting to a different full node, or running the wallet in "light" mode.

[Read more about network-level Bitcoin wallet privacy here.](/using-wasabi/NetworkLevelPrivacy.md)

![](/StatusBarBackendConnected.png)

Back to Wasabi.
After loading the wallet, the user can generate a receive address.
Some important design choices were made here.
First, Wasabi had to be a Segregated Witness only wallet, so the registration of unconfirmed CoinJoin outputs into a new CoinJoin round is done to prevent malleability attacks.

However, the developers of Wasabi decided to make the wallet native SegWit (bech32) only, not supporting wrapped SegWit.
This way, the backend server can leverage this, and only generate filters regarding bech32 addresses.
This makes Wasabi's filter size a few megabytes today, instead of >4GB.

At first glance, this may be seen as hazardous to privacy.

> However, since Wasabi user UTXOs can be recognized as such by the huge CoinJoins that only Wasabi does anyway, there is minimal-to-no additional privacy loss.

Author: ```suggestion
However, since UTXOs CoinJoined by using Wasabi can be recognized as such by their huge anonymity sets that only Wasabi creates, there is minimal-to-no additional privacy loss.
```

Author: @yahiheb I don't think it's better to use CoinJoin twice in the same sentence.  How about this?

Reviewer: ACK

However, technology is improving quickly, thus, timing has special importance in this matter. [https://github.com/zkSNACKs/Meta/issues/9](https://github.com/zkSNACKs/Meta/issues/9)

### Web Wallet

The question of a web-wallet is also something to think about.
However, it may not be possible to build a network-analysis resistant web wallet, nor to build a secure web wallet in general.
Nevertheless [this question](https://github.com/zkSNACKs/Meta/issues/20) deserves more thought.

### Hardware Wallet

Wasabi has [hardware wallet integration](https://github.com/zkSNACKs/WalletWasabi/pull/134) however in this mode CoinJoining is not possible.

### Bitcoin Core

A wallet that is connected to a full node is the ultimate way of minimizing trust.
Wasabi now offers 3 different ways to connect to a full node:

- If you are already running a full node on the same computer, Wasabi will automatically detect and connect to it.
- You may connect to a trusted remote node via the `Settings` tab.
- Wasabi includes an optional, integrated full node, which can be enabled in the `Settings` tab.

### Daemon/API

Wasabi now offers an RPC interface.

> Full information can be found [here.](/using-wasabi/RPC.html)

Reviewer 2: ```suggestion
Full information can be found [here.](/using-wasabi/RPC.md)
```
